### PR TITLE
ClangFnMapGen rewritten to give output to stdout. Unneeded files, suc…

### LINKED ISSUE #117: refactor clangfnmap tool
-clangfnmap tool should give output to stdout only (and not into files)
-clangfnmap should only print mangled names of defined functions and the location of definition and nothing else.

### PR DESCRIPTION
**Pull request by dkrupp**
clangfnmap refactored.
fixes #117

There is still one todo: implement the locking of externalFnMap.txt in ctu-build.py.

Also please implement tests for clangfnmapgen

### CODE REVIEW

File: tools/clang-func-mapping/ClangFnMapGen.cpp [hunk 1]
```diff
@@ -182,19 +130,20 @@ class MapFunctionNamesAction : public ASTFrontendAction {
...
+
 int main(int argc, const char **argv) {
   // Print a stack trace if we signal out.
   sys::PrintStackTraceOnErrorSignal(argv[0], false);
   PrettyStackTraceProgram X(argc, argv);
 
   SmallVector<std::string, 4> Sources;
+  const char *overview =
+      "\nThis tool collects the mangled name and location of definition "
+      "of all functions defined in the given compilation command.\n";
```
Comment (Xazax-hun) [inline tools/clang-func-mapping/ClangFnMapGen.cpp]: This tool dumps the definitions that available in the C/C++ files (omitting definitions from headers). Not definitions from the compilation commands.

File: tools/ctu-analysis/ctu-build.py [hunk 2]
```diff
@@ -144,17 +144,35 @@ def generate_ast(params):
 def map_functions(params):
     command, sources, directory, clang_path, ctuindir = params
     args = get_command_arguments(command)
+    print "map_funcions command:"+command
```
Comment (Xazax-hun) [inline tools/ctu-analysis/ctu-build.py]: Use function like print.

Comment (gerazo) [inline tools/ctu-analysis/ctu-build.py]: Or better, use logging! Logs are set to be emitted only on giving -v option.

File: tools/ctu-analysis/ctu-build.py [hunk 3]
```diff
@@ -144,17 +144,35 @@ def generate_ast(params):
...
     funcmap_command.extend(sources)
     funcmap_command.append('--')
     funcmap_command.extend(args)
     funcmap_command_str = ' '.join(dir_command) + \
                           " && " + ' '.join(funcmap_command)
     logging.info(funcmap_command_str)
-    subprocess.call(funcmap_command_str, shell=True)
-
+    print "Calling function map:"+funcmap_command_str
+    output=[]
```
Comment (gerazo) [inline tools/ctu-analysis/ctu-build.py]: There are some pep8 breaches in this block.

File: tools/clang-func-mapping/ClangFnMapGen.cpp [hunk 4]
```diff
@@ -113,34 +80,30 @@ void MapFunctionNamesConsumer::handleDecl(const Decl *D) {
...
+          CurrentFileName = Path;
+          free(Path);
+        }
+
+        switch (FD->getLinkageInternal()) {
+        case ExternalLinkage:
+        case VisibleNoLinkage:
+        case UniqueExternalLinkage:
+          if (SM.isInMainFile(Body->getLocStart())) {
+            // DefinedFuncsStr << "!";
```
Comment (gerazo) [inline tools/clang-func-mapping/ClangFnMapGen.cpp]: I think we can remove this comment. Unnecessary load for the future generations.

File: tools/clang-func-mapping/ClangFnMapGen.cpp [hunk 5]
```diff
@@ -149,24 +112,9 @@ void MapFunctionNamesConsumer::handleDecl(const Decl *D) {
       handleDecl(D);
 }
 
-bool MapFunctionNamesConsumer::isCLibraryFunction(const FunctionDecl *FD) {
```
Comment (gerazo) [inline tools/clang-func-mapping/ClangFnMapGen.cpp]: Wow, a totally useless function. Nice catch.

**Pull request closed without merging**